src/beamlime/applications/_random_data_providers.py

```diff
@@ -24,108 +18,37 @@
 RandomEvents = NewType("RandomEvents", List[sc.DataArray])
 
 
-def provide_rng(random_seed: RandomSeed) -> RNG:
-    from numpy.random import default_rng
-
-    return default_rng(random_seed)
-
+DetectorNumberCands = NewType("DetectorNumberCands", List[int])
 
-def calculate_event_per_frame(
-    frame_rate: FrameRate, event_rate: EventRate
-) -> EventFrameRate:
-    return EventFrameRate(int(event_rate / frame_rate))
 
+class EV44(TypedDict):
+    reference_time: list[float] | np.ndarray
+    reference_time_index: list[int] | np.ndarray
+    time_of_flight: list[float] | np.ndarray
+    pixel_id: list[int] | np.ndarray
 
-def provide_time_coords(
-    rng: RNG,
-    ef_rate: EventFrameRate,
-    ref_time: Optional[ReferenceTimeZero] = None,
-) -> TimeCoords:
-    ref_time = ref_time or ReferenceTimeZero(13620492**11)
-    et_zero = sc.array(dims=["event"], values=[ref_time] * ef_rate, unit='ns')
-    et_offset = sc.array(
-        dims=["event"], values=rng.random((ef_rate,)) * 800 + 200, unit='ns'
-    )
 
-    return TimeCoords({"event_time_zero": et_zero, "event_time_offset": et_offset})
+RandomEV44Generator = Generator[EV44, Any, Any]
 
 
-def provide_time_coords_generator(
-    rng: RNG, ef_rate: EventFrameRate, num_frames: NumFrames, frame_rate: FrameRate
-) -> Generator[TimeCoords, Any, Any]:
-    for i_frame in range(num_frames):
-        ref_time = ReferenceTimeZero(
-            int(13620492e11 + (i_frame / frame_rate) * 10**9)
-        )
-        yield provide_time_coords(rng, ef_rate, ref_time)
-
-
-def provide_dummy_counts(ef_rate: EventFrameRate) -> DetectorCounts:
-    return DetectorCounts(sc.ones(sizes={"event": ef_rate}, unit='counts'))
+def random_ev44_generator(
+    detector_numbers: DetectorNumberCands,
+) -> Generator[EV44, Any, Any]:
+    """Randomly select detector numbers (pixel ids) and generate events per frame."""
+    rng = default_rng(123)
+    event_rate = 10_000
+    frame_rate = 14
+    ef_rate = int(event_rate / frame_rate)
 
-
-def provide_random_pixel_id_generator(
-    rng: RNG, ef_rate: EventFrameRate, num_pixels: NumPixels, num_frames: NumFrames
-) -> Generator[RandomPixelId, Any, Any]:
-    for _ in range(num_frames):
-        yield RandomPixelId(
-            sc.array(
-                dims=["event"],
-                values=rng.integers(low=0, high=num_pixels, size=ef_rate),
-                unit=sc.units.dimensionless,
-            )
+    et_zero = ReferenceTimeZero(13620492**11)  # No reason
+    while True:
+        cur_event_number = int(
+            ef_rate * (rng.integers(99, 101) / 100)
+        )  # 1% of fluctuation
+        yield EV44(
+            reference_time=[et_zero],
+            reference_time_index=[0],
```

Comment:
Good question. Will the stream use the time index to refer to the history (then this should be something like the cumsum of the number of events in all previous messages since the run-start), or only the current message?

Reply:
I started a thread of QnA in #168 to collect questions and remind us to document them.